Add Cloud compatibility notes to guides

ptgott opened the pull request:
Add Details boxes to some guides that do not mention Teleport
Cloud, indicating how each guide relates to Teleport Cloud users.

ptgott committed:
Add Cloud compatibility notes to guides

Add Details boxes to some guides that do not mention Teleport
Cloud, indicating how each guide relates to Teleport Cloud users.

zmb3 commented on docs/pages/application-access/guides/api-access.mdx:
```diff
@@ -8,6 +8,10 @@ description: How to access REST APIs with Teleport Application Access.
 Teleport Application Access can be used to access applications' (REST or Teleport's own gRPC) APIs with
 tools like [curl](https://man7.org/linux/man-pages/man1/curl.1.html) or Postman.
 
+<Details scopeOnly={true} scope={["cloud"]} title="Teleport Cloud Compatibility">
```
Is the idea that we're going to put a note like this on every guide that is compatible with cloud?

1-2 years from now I will expect that 90% or more of our guides work on cloud, won't this get repetitive? Would it be easier to assume cloud compatibility and only call out cases that don't work on cloud?

ptgott committed:
Respond to PR feedback

Removes Details boxes unless necessary to warn users of
incompatibility with Teleport Cloud.

Also makes minor stylistic tweaks.

ulysseskan commented on docs/pages/getting-started/linux-server.mdx:
````diff
@@ -206,15 +214,15 @@ $ tsh ls
 ```
 
-Bootstrap a new node:
+Bootstrap a new Node:
````
I probably wouldn't capitalize node unless it were part of a title.

ulysseskan commented on docs/pages/getting-started/linux-server.mdx:
````diff
@@ -206,15 +214,15 @@ $ tsh ls
 ```
 
-### Add a node to the cluster
+### Add a Node to the cluster
````
If you're going to capitalize node in this title, you might want to capitalize cluster as well.

ptgott replied:
See my note re: "Node" above

ptgott commented:
@zmb3 @ulysseskan Thanks for reviewing—this should be ready for another look.

ulysseskan reviewed (approved):
Looks good to me, Paul.  Thanks!

russjones commented:
@ptgott Looks like this is approved, can we merge?
